File: DAIDALUS/C++/src/NoneUrgencyStrategy.cpp
/*
 * UrgencyStrategy.cpp
 *
 * This strategy always returns an INVALID aircraft.
 *
 */

#include "NoneUrgencyStrategy.h"

namespace larcfm {

TrafficState NoneUrgencyStrategy::mostUrgentAircraft(Detection3D* detector, const OwnshipState& ownship, const std::vector<TrafficState>& traffic, double T) {
  return TrafficState::INVALID;
}

UrgencyStrategy* NoneUrgencyStrategy::copy() const {
  return new NoneUrgencyStrategy();
}

}

File: DAIDALUS/Java/src/gov/nasa/larcfm/ACCoRD/NoneUrgencyStrategy.java
package gov.nasa.larcfm.ACCoRD;

import java.util.List;

/* 
 * This strategy always returns an INVALID aircraft.
 */
public class NoneUrgencyStrategy implements UrgencyStrategy {

  /**
   * @return INVALID aircraft
   */

  public TrafficState mostUrgentAircraft(Detection3D detector, OwnshipState ownship, List<TrafficState> traffic, double T) {
    return TrafficState.INVALID;  
  }

  public UrgencyStrategy copy() {
    return new NoneUrgencyStrategy();
  }
 
}
